### PR TITLE
codecoverage/backend: Fix bug to load a report for a changeset.

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: src/codecoverage/backend/tests/test_gcp.py [hunk 1]
```diff
@@ -153,13 +153,31 @@ def test_closest_report(mock_cache, mock_hgmo):
...
+    # ... and the revision on the push itself
+    revision = '994{}'.format(uuid.uuid4().hex[3:])
+    assert mock_cache.find_closest_report('myrepo', revision) == (report_rev, 994)
+
+    # But not for revisions after the push
+    revision = '995{}'.format(uuid.uuid4().hex[3:])
+    with pytest.raises(Exception) as e:
+        mock_cache.find_closest_report('myrepo', revision)
+    assert str(e.value) == 'No report found'
+
```
Comment (marco-c) [inline src/codecoverage/backend/tests/test_gcp.py]: Also check that 989 -> 990, 990 -> 990

File: src/codecoverage/backend/codecoverage_backend/gcp.py [hunk 2]
```diff
@@ -241,18 +241,18 @@ def find_closest_report(self, repository, changeset):
             self.ingest_pushes(repository, min_push_id=push_id-1, nb_pages=1)
 
         # Load report from that push
-        return self.find_report(repository, min_push_id=push_id)
+        return self.find_report(repository, max_push_id=push_id, min_push_id='+inf', reverse=False)
```
Comment (marco-c) [inline src/codecoverage/backend/codecoverage_backend/gcp.py]: It's a bit confusing that max_push_id < min_push_id

File: src/codecoverage/backend/codecoverage_backend/gcp.py [hunk 3]
```diff
@@ -201,11 +201,11 @@ def store_push(self, repository, push_id, changesets, date):
 
         logger.info('Stored new push data', push_id=push_id)
 
-    def find_report(self, repository, min_push_id=None, max_push_id=None):
+    def find_report(self, repository, **kwargs):
```
Comment (marco-c) [inline src/codecoverage/backend/codecoverage_backend/gcp.py]: Nit: I prefer an explicit list of parameters

File: src/codecoverage/backend/codecoverage_backend/gcp.py (hunk 2 above)
Comment (La0) [reply src/codecoverage/backend/codecoverage_backend/gcp.py]: Reverse order by default to get the history & latest easily

**Pull request merged**